tests/unit/demography/test_t_model_functions.py

```diff
@@ -727,6 +661,54 @@ def test_calculate_dbh_from_height_edge_cases():
     assert np.all(np.isnan(dbh) == np.array([[0, 0], [0, 0], [0, 0], [1, 0], [1, 1]]))
 
 
+@pytest.mark.parametrize(
+    argnames="at_dbh, outcome, excep_msg",
+    argvalues=[
+        pytest.param(np.array(1), does_not_raise(), None, id="pass_0D"),
+        pytest.param(np.ones(1), does_not_raise(), None, id="pass_1D_scalar"),
+        pytest.param(np.ones(3), does_not_raise(), None, id="pass_1D_row"),
+        pytest.param(np.ones((1, 3)), does_not_raise(), None, id="pass_2D_row"),
+        pytest.param(np.ones((4, 1)), does_not_raise(), None, id="pass_2D_col"),
+        pytest.param(np.ones((4, 3)), does_not_raise(), None, id="pass_2D_full"),
+        pytest.param(
+            np.ones(4),
+            pytest.raises(ValueError),
+            "The array shapes of the trait (3,) and size (4,) arguments",
```

Comment:
It is but its enough to confirm the expected raise and the test uses `startswith`. I guess I'm just trying to avoid extra lines in the test parameters.